### PR TITLE
Added retry attempt if context switched too early

### PR DESCRIPTION
**Pull request by alexhurs**
These changes are intended to help to fix the following problem:

sometimes context is switched too early and UI elements are searched on the wrong page

### CODE REVIEW

**Review by SergeyRuzin — APPROVED** [review]

File: src/Behavioral.Automation/Services/AutomationIDProvider.cs [hunk 1]
```diff
@@ -10,22 +13,41 @@ namespace Behavioral.Automation.Services
...
         {
             ParseCaption(caption, out var name, out var type);
-            return _scopeContextRuntime.FindControlDescription(type, name);
+
+            var controlDescription = _scopeContextRuntime.FindControlDescription(type, name);
+
+            if (controlDescription != null) 
+                return controlDescription;
+
+            Thread.Sleep(500);
```
Comment (andrey-brovko) [inline src/Behavioral.Automation/Services/AutomationIDProvider.cs:44]: What do you think, one attempt will be enough?

Comment (alexhurs) [reply src/Behavioral.Automation/Services/AutomationIDProvider.cs:44]: I performed several test runs (up to 800 tests for each) and context related exceptions had never been raised, so I think 500 ms is enough

File: src/Behavioral.Automation/Services/AutomationIDProvider.cs [hunk 2]
```diff
@@ -10,22 +13,41 @@ namespace Behavioral.Automation.Services
...
+            _scopeContextManager.SwitchToCurrentUrl(new Uri(_driverService.CurrentUrl));
+
+
```
[the remark this answers is not in the document]
Comment (alexhurs) [reply src/Behavioral.Automation/Services/AutomationIDProvider.cs]: extra line removed

File: src/Behavioral.Automation/Services/AutomationIDProvider.cs (hunk 1 above)
Comment (andrey-brovko) [reply src/Behavioral.Automation/Services/AutomationIDProvider.cs:44]: Ok. In case of a similar issue, we can add several attempts with a 500 ms delay.

**Review by andrey-brovko — APPROVED** [review]

**Pull request merged**